Update AWS and GCP README and add docs

Edwin-Pau opened the pull request:
-Created table of contents for both AWS and GCP README.md

-Added a section under requirements to setup SSH keys

-Added a new section for 'Troubleshooting'

-Added a new document for location of logs 'debugging.md'

-Added a new document for setting up SSH keys

Signed-off-by: Edwin-Pau <epau@teradici.com>

Commit:
Update AWS and GCP README and add docs

-Created table of contents for both AWS and GCP README.md

-Added a section under requirements to setup SSH keys

-Added a new section for 'Troubleshooting'

-Added a new document for location of logs 'debugging.md'

-Added a new document for setting up SSH keys

Signed-off-by: Edwin-Pau <epau@teradici.com>

shermanyin commented on docs/debugging.md:
```diff
@@ -0,0 +1,90 @@
+# Debugging
```
ToC?

shermanyin replied:
Maybe call this file troubleshooting

shermanyin commented on docs/debugging.md:
````diff
@@ -0,0 +1,90 @@
+### Connecting to CentOS Workstations
+One way to access CentOS Workstations is to use the Connector as a bastion host. Execute the following command to first SSH into the Connector:
+
+```ssh -A <login-user>@<cac-public-ip>``` 
````
add -i to indicate which ssh key to use. explain to use the private key corresponding to cac_admin_ssh_pub_key_file or centos_admin_ssh_pub_key_file for GCP or admin_ssh_pub_key_file for AWS.
BTW we should make it the same for AWS and GCP - use the same key for both cac and centos

Edwin-Pau replied:
I've added this on my list of to do's.

shermanyin commented on docs/debugging.md:
````diff
@@ -0,0 +1,90 @@
+
+```
+Computer: <domain-controller-public-ip>
+User: <login-user>
````
say "Administrator" or "cam_admin" instead of "login-user", also use the corresponding password.

shermanyin commented on docs/debugging.md:
````diff
@@ -0,0 +1,90 @@
+
+```
+Computer: <win-internal-ip>
+User: <login-user>
````
specify what login-user

shermanyin commented on docs/debugging.md:
```diff
@@ -0,0 +1,90 @@
+| VM Instance      | Login User    | Log File Path                               | Description                                                 |
+| :--------------- | :------------ | :------------------------------------------ | :---------------------------------------------------------- |
+| cac              | ubuntu        | /var/log/syslog                             | Detailed system log for startup and provisioning            |
+|                  |               | /var/log/teradici/provisioning.log          | Minimal log for provisioning script                         |
```
is it still minimal log? thought we added set -x to show everything?

Edwin-Pau replied:
That commit (improve provisioning scripts branch) hasn't been merged yet, but I've changed this table to show detailed logs instead so I don't have to go back to it later.

shermanyin commented on docs/README-aws.md:
```diff
@@ -133,3 +158,11 @@ Be sure to SSH into the Local License Server (LLS), possibly using a Cloud Acces
 
 ![aws-lb-connectors-lls diagram](aws-lb-connectors-lls.png)
+
+## Troubleshooting
```
Might as well just have one line pointing to the other file.

shermanyin commented on docs/ssh-key-pair-setup.md:
```diff
@@ -0,0 +1,18 @@
+
+## Creating a SSH key pair on Windows machines
+- Create a key pair using PuTTYgen. Please refer to https://www.ssh.com/ssh/putty/windows/puttygen for instructions to create SSH keys. 
+- Once the public / private key pair is created, the SSH2 public key format must be converted to a OpenSSH format for Terraform. Please refer to https://tutorialinux.com/convert-ssh2-openssh/ for detailed instructions. 
```
we can put the reference after our instructions in case they need to know more, so this section is:
1. download puttygen from ...
2. create sshkey by clicking "Generate" with "RSA" selected
3. run the following command in command prompt to convert ....
'command to run'
For details, refer to
- link to puttygen
- link to convert key

shermanyin commented on docs/ssh-key-pair-setup.md:
````diff
@@ -0,0 +1,18 @@
+```ssh-keygen -t rsa```
+
+- Enter file in which to save the key (home/user/.ssh/id_rsa), press enter to save in the default location.
+- Enter passphrase (empty for no passphrase), press enter for no passphrase.
````
i haven't tried, but make sure our deployment works with a password-protected ssh key.  i think it should.

Edwin-Pau replied:
Confirmed it works with password protected priv key

Commit:
Update documentation

Updated and streamlined the new documentation with table of contents
and better instructions.

Signed-off-by: Edwin-Pau <epau@teradici.com>

shermanyin commented on docs/README-gcp.md:
```diff
@@ -172,3 +201,6 @@ A simple deployment of one Domain Controller, intended for testing Domain Contro
 Directory, DNS, LDAP-S.  One AD Service Account is also created.
+
+## Troubleshooting
+Please visit the [Troubleshooting](/docs/troubleshooting) page for further instructions.
```
dead link. missing .md

shermanyin commented on docs/README-aws.md:
```diff
@@ -133,3 +156,6 @@ Be sure to SSH into the Local License Server (LLS), possibly using a Cloud Acces
 ![aws-lb-connectors-lls diagram](aws-lb-connectors-lls.png)
+
+## Troubleshooting
+Please visit the [Troubleshooting](/docs/troubleshooting) page for further instructions.
```
deadlink. missing .md

Commit:
Fix deadlinks to troubleshooting.md

Added .md to fix two deadlinks to troubleshooting.md from README-aws.md
and README-gcp.md.

Signed-off-by: Edwin-Pau <epau@teradici.com>
